app/services/order_syncer.rb

```diff
@@ -106,4 +110,22 @@ def force_ship_address_required?(order)
       order.ship_address[attr] == distributor_address[attr]
     end
   end
+
+  def select_shipping_method_for_shipment(shipment, shipping_method)
+    shipping_rate = find_or_create_shipping_rate_for_shipping_method(shipment, shipping_method)
+    shipment.update_attributes(selected_shipping_rate_id: shipping_rate.id)
```

Comment:
should be shipment.selected_shipping_rate_id=(shipping_rate.id)

Reply:
@luisramos0 Using `update_attributes` also works because `selected_shipping_rate_id` is `attr_accessible`: https://github.com/openfoodfoundation/spree/blob/2-0-4-stable/core/app/models/spree/shipment.rb#L19 And specs would fail if in the future it won't allow mass assignment anymore.